packages/grafana-data/src/transformations/transformers/reduce.ts

```diff
@@ -222,9 +221,10 @@ export function reduceFields(data: DataFrame[], matcher: FieldMatcher, reducerId
         });
         for (const reducer of reducers) {
           const value = results[reducer];
+          const type = getFieldType(reducer, field);
           const copy = {
             ...field,
-            type: guessFieldTypeFromValue(value),
+            type: type === FieldType.other ? guessFieldTypeFromValue(value) : type,
```

Comment:
```suggestion
            type: getFieldType(reducer, field),
```
all and unique will always produce an array (and that is other)